src/commands/ideCommand.ml

```diff
@@ -0,0 +1,79 @@
+(**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the "flow" directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ *)
+
+(***********************************************************************)
+(* flow ide command *)
+(***********************************************************************)
+
+open CommandUtils
+module Prot = ServerProt.Persistent_connection_prot
+
+let spec = {
+  CommandSpec.
+  name = "ide";
+  doc = "Starts a persistent connection to the server. Currently in development and highly unstable";
+  usage = Printf.sprintf
+    "Usage: %s ide\n\n\
+      Starts a persistent connection to the server\n"
+      CommandUtils.exe_name;
+  args = CommandSpec.ArgSpec.(
+    empty
+    |> server_flags
+    |> root_flag
+    (* TODO consider using strip_root |> strip_root_flag *)
+    (* TODO use this somehow? |> verbose_flags *)
+  )
+}
+
+let handle_server_message fd =
+  let (message : Prot.response) =
+    Marshal_tools.from_fd_with_preamble fd
+  in
+  let Prot.Errors errors = message in
+  let error_count = Errors.ErrorSet.cardinal errors in
+  (* TODO we will use JSON RPC at some point in the near future. this simple message format is just
+  for ease of prototyping *)
+  Printf.printf "Error count: %d" error_count;
+  print_newline ()
+
+let send_server_request fd msg =
+  Marshal_tools.to_fd_with_preamble fd (msg: Prot.request)
+
+let handle_stdin_message server_fd =
+  (* TODO use buffered_line_reader *)
+  let line = input_line stdin in
+  (* TODO we will use JSON RPC at some point in the near future. this simple message format is just
+  for ease of prototyping *)
+  match line with
+    | "subscribe" ->
+        send_server_request server_fd Prot.Subscribe
+    | _ -> print_endline ("not recognized: " ^ line)
+
+let main option_values root () =
+  let root = CommandUtils.guess_root root in
+  let ic, oc = connect option_values root in
+  ServerProt.cmd_to_channel oc ServerProt.CONNECT;
+  let stdin_fd = Unix.descr_of_in_channel stdin in
+  let ic_fd = Timeout.descr_of_in_channel ic in
+  let oc_fd = Unix.descr_of_out_channel oc in
+  while true do
+    let readable_fds, _, _ = Unix.select [stdin_fd; ic_fd] [] [] 1.0 in
+    match readable_fds with
```

Comment:
Yeah, but the one we don't deal with here will just come up again in the next select call. I don't see a reason to complicate it